plugins/module_utils/hcloud.py

```diff
@@ -32,6 +35,30 @@ def __init__(self, module, represent):
             module.fail_json(msg=missing_required_lib("python-dateutil"))
         self._build_client()
 
+    def fail_json_hcloud(self, exception, msg=None, params=None, **kwargs):
+        last_traceback = traceback.format_exc()
```

Comment:
The exception is an object, and it is risky to return it as value through `module.fail_json` (we don't know how it will be printed). Instead, we want a well formatted trace as string for ansible to print in verbose mode: https://docs.ansible.com/ansible/latest/reference_appendices/common_return_values.html#exception